### ISSUE TITLE
acs-engine quick-create ...

### ISSUE THREAD

**Issue by khenidak** [Azure/acs-engine#3378]
We have talked multiple times with regards to optimized creation process (cc: @jessfraz @jackfrancis @cpuguy83, @CecileRobertMichon ) - The current process is:
1. **Lengthy** : copy api-model,  modify values, do `acs-engine generate` then deploy. Each step is prone to errors (spelling issues, copy/paste etc).
2. The current process is **not optimized the best practices** (turn on MSI, use premium disks etc). 

#### Proposal: 
1. Establish a well know golden api-model (as part of the binary). That golden model encompasses all the best practices that we learned and learning.  The api-model will also be updated as part of the release as needed.
2. Create a new CLI command `acs-engine quick-create --resource-group --location --master-count {n} --agentpool-count {n} --node-per-pool {n} --overrides [{key:val}] --output-directory {path}`  calling this command will:

1. Generate the needed api-model + ARM template
2. Create resource-group
3. Perform deployment

> It is assumed that some of the advanced feature (such as BYO VNET) will not supported by `quick-create`

@cpuguy83 is willing to carry this forward. 

comments/questions?

**Comment by cpuguy83** [comment]
Not sure how I feel about the `--overrides` flag. It's feels a bit too loosey-goosey.
What if it can generate a golden template that users can edit for overrides?

**Comment by khenidak** [comment]
I am game not to have overrides at all.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by cpuguy83** [comment]
How about `acs-engine deploy --with-builtin-model --builtin-model-name v0` where, for now, the new flags are mutually exclusive with `--api-model` and the version flag can be used to deploy a specific version but the default is the latest version.

Alternatively `--api-model` could default to the value of `builtin` (thereby also making it a reserved word). This would still include the model name flag as well.

**Comment by serbrech** [comment]
Note that we have the --set flag implemented by @jcorioland that might solve your `--override` idea 

merged PR : #2787 
[--set flag docs](https://github.com/jcorioland/acs-engine/blob/master/docs/kubernetes/deploy.md#step-4-generate-the-templates)

**Comment by cpuguy83** [comment]
Another thought here:

```
$ acs-engine create [basic opts to configure model] > model.json
<edit and/or commit model>
$ acs-engine deploy --api-model model.json
```

As well as a simple

```
$ acs-engine create [basic opts to configure model] --deploy
```

These could also be flipped to deploy by default and have a flag to output the model that can be consumed by the `deploy` subcommand.
